packages/x-data-grid/src/components/cell/GridCell.tsx

```diff
@@ -207,7 +207,8 @@ const GridCell = React.forwardRef<HTMLDivElement, GridCellProps>(function GridCe
       result.api = apiRef.current;
       return result;
     },
-    objectShallowCompare,
+    undefined,
+    objectShallowCompare as (a: unknown, b: unknown) => boolean,
```

Comment:
Should we type `objectShallowCompare` directly? That function is just a re-export anyway, it's meant to be used for this case, it would be cool if it had the proper type.